### PR TITLE
Add fixes for TOMAS simulations in GCHP

### PR DESCRIPTION
**Pull request by msulprizio**
### Name and Institution (Required)

Name: Melissa Sulprizio
Institution: Harvard / GCST

### Describe the update

This pull request supersedes https://github.com/geoschem/geos-chem/pull/2116, which included fixes for running TOMAS in GCHP.  This pull request also includes:

1. Removing hardcoded TOMAS species from GCHP's HISTORY.rc.fullchem and adding them only for TOMAS simulations (in script `setupConfigFiles.sh`)
2. Updating the sample restart files for TOMAS simulations in GCHP

This should be merged alongside:

- https://github.com/geoschem/HEMCO/pull/259

### Expected changes

These updates produce zero differences with respect to the full-chemistry benchmark simulation. They now allow all GCHP integrations (including TOMAS15) to pass:

```
Execution tests:
------------------------------------------------------------------------------
gchp_fullchem_merra2................................Execute Simulation....PASS
gchp_fullchem_merra2_benchmark......................Execute Simulation....PASS
gchp_fullchem_merra2_RRTMG..........................Execute Simulation....PASS
gchp_fullchem_merra2_TOMAS15........................Execute Simulation....PASS
gchp_tagO3_merra2...................................Execute Simulation....PASS
gchp_TransportTracers_merra2........................Execute Simulation....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 6
Execution tests failed: 0
Execution tests not yet completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

### Related Github Issue(s)

- https://github.com/geoschem/geos-chem/issues/1741

### CODE REVIEW

File: run/GCClassic/createRunDir.sh [hunk 1]
```diff
@@ -1045,7 +1045,7 @@ fi
 
 #--------------------------------------------------------------------
 # Copy sample restart file to run directory
-# Bash functions used here are from ../shared/setupForRestarts.sh
+# Bash functions used here are from ./setupForRestarts.sh
 #--------------------------------------------------------------------
```
Comment (yantosca) [inline run/GCClassic/createRunDir.sh:1049]: Have you also updated the restart file path in `./run/shared/download_data.yml`?

File: GeosCore/tomas_mod.F90 [hunk 2]
```diff
@@ -524,41 +524,41 @@ SUBROUTINE AEROPHYS( Input_Opt, State_Chm, State_Grid, State_Met, &
...
+!                ionrate=ionrate + ( soil_ions( lev   ) * weight + &
+!                        soil_ions( lev-1 ) * (1.e+0_fp-weight) )
+!             endif
+!          endif
+!
+!       ELSE
+!          ionrate = 0.e+0_fp
+!       ENDIF
+!
+!       if(ionrate .le. 1.501) ionrate = 1.501
```
Comment (yantosca) [inline GeosCore/tomas_mod.F90:561]: Can this code be deleted if no longer used?

File: GeosCore/tomas_mod.F90 [hunk 3]
```diff
@@ -6355,16 +6507,17 @@ SUBROUTINE INIT_TOMAS( Input_Opt, State_Chm, State_Grid, RC )
...
-    ENDIF
+    !IF ( TRIM(State_Grid%GridRes) == '4.0x5.0' ) THEN
+    !   SGCTSCALE = 10.*3600.  ! 10 hours
+    !ELSE IF ( TRIM(State_Grid%GridRes) == '2.0x2.5' ) THEN
+    !   SGCTSCALE = 5.*3600.
+    !ELSE IF ( TRIM(State_Grid%GridRes) == '0.5x0.625' ) THEN
+    !   SGCTSCALE = 1.*3600.
+    !ELSE IF ( TRIM(State_Grid%GridRes) == '0.25x0.3125' ) THEN
+    !   SGCTSCALE = 0.5*3600.
+    !ENDIF
```
Comment (yantosca) [inline GeosCore/tomas_mod.F90:6520]: Can this be deleted?

File: GeosCore/sulfate_mod.F90 [hunk 4]
```diff
@@ -1045,17 +1045,18 @@ SUBROUTINE SRCSF30( Input_Opt, State_Grid, State_Met, State_Chm, TC2, RC )
...
-    ELSE IF ( TRIM(State_Grid%GridRes) == '0.5x0.625' ) THEN
-       TSCALE = 1.*3600.
-    ELSE IF ( TRIM(State_Grid%GridRes) == '0.25x0.3125' ) THEN
-       TSCALE = 0.5*3600.
-    ENDIF
+    !!ELSE IF ( TRIM(State_Grid%GridRes) == '0.5x0.625' ) THEN
+    !   TSCALE = 1.*3600.
+    !ELSE IF ( TRIM(State_Grid%GridRes) == '0.25x0.3125' ) THEN
+    !   TSCALE = 0.5*3600.
+    !ENDIF
```
Comment (yantosca) [inline GeosCore/sulfate_mod.F90:1059]: Can the commented-out code be deleted?

**Review by yantosca — CHANGES_REQUESTED** [review]
Hi @msulprizio, thanks for this PR.  Mostly looks good but I did note a couple of minor points.

File: GeosCore/sulfate_mod.F90 (hunk 4 above)
Comment (BettyCroft) [reply GeosCore/sulfate_mod.F90:1059]: Hi @yantosca - would it be possible to leave these commented sections in place for now? The idea being that they could serve as a basis if we decide to make future developments to restore functionality of TOMAS' SGCOAG switch by moving the switch and associated parameters into input options?

**Comment by yantosca** [comment]
@BettyCroft: I'm OK with this.

**Comment by BettyCroft** [comment]
Thank you @yantosca!

File: run/GCClassic/createRunDir.sh (hunk 1 above)
Comment (msulprizio) [reply run/GCClassic/createRunDir.sh:1049]: No, the GCClassic TOMAS restart files are not updated here. We only add TOMAS restarts for GCHP.

File: GeosCore/sulfate_mod.F90 (hunk 4 above)
Comment (msulprizio) [reply GeosCore/sulfate_mod.F90:1059]: Yes @BettyCroft. We will leave the commented out code for now.

File: GeosCore/tomas_mod.F90 [hunk 5]
```diff
@@ -524,41 +524,41 @@ SUBROUTINE AEROPHYS( Input_Opt, State_Chm, State_Grid, State_Met, &
...
-                        soil_ions( lev-1 ) * (1.e+0_fp-weight) )
-             endif
-          endif
-
-       ELSE
-          ionrate = 0.e+0_fp
-       ENDIF
-
-       if(ionrate .le. 1.501) ionrate = 1.501
+       ionrate = 10. ! set as constant now !!jrp, bc 18/12/23 and comment out below
```
[the remark this answers is not in the document]
Comment (msulprizio) [reply GeosCore/tomas_mod.F90]: Now fixed in [ca65457](https://github.com/geoschem/geos-chem/pull/2136/commits/ca6545718e9dd476259447768f3ada1b116181b6)

File: GeosCore/sulfate_mod.F90 [hunk 6]
```diff
@@ -1045,17 +1045,18 @@ SUBROUTINE SRCSF30( Input_Opt, State_Grid, State_Met, State_Chm, TC2, RC )
...
+    
+     ! comment out if shut off subgridcoag bc,14/12/23
     ! Define subgrid coagulation timescale (win, 10/28/08)
-    IF ( TRIM(State_Grid%GridRes) == '4.0x5.0' ) THEN
-       TSCALE = 10.*3600.  ! 10 hours
-    ELSE IF ( TRIM(State_Grid%GridRes) == '2.0x2.5' ) THEN
+    !IF ( TRIM(State_Grid%GridRes) == '4.0x5.0' ) THEN
+    !   TSCALE = 10.*3600.  ! 10 hours
+    !ELSE IF ( TRIM(State_Grid%GridRes) == '2.0x2.5' ) THEN
        TSCALE = 5.*3600.
```
[the remark this answers is not in the document]
Comment (msulprizio) [reply GeosCore/sulfate_mod.F90]: Now fixed in [ca65457](https://github.com/geoschem/geos-chem/pull/2136/commits/ca6545718e9dd476259447768f3ada1b116181b6)

**Review by yantosca — APPROVED** [review]
Good to merge!  Thanks @msulprizio

**Comment by msulprizio** [comment]
This update passes all GCClassic and GCHP integration tests. I have also confirmed zero differences in fullchem benchmark simulations from these updates.

**Pull request merged**